File: src/main/java/com/thetaris/goldprice/config/WsdlConfig.java
package com.thetaris.goldprice.config;

import com.thetaris.goldprice.WsdlProperties;
import com.thetaris.goldprice.generated.GetCurrentGoldPrice;
import com.thetaris.goldprice.generated.ObjectFactory;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
@EnableConfigurationProperties(WsdlProperties.class)
public class WsdlConfig {

    private final WsdlProperties wsdlProperties;

    public WsdlConfig(WsdlProperties wsdlProperties) {
        this.wsdlProperties = wsdlProperties;
    }

    @Bean
    public ObjectFactory objectFactory() {
        return new ObjectFactory();
    }

    @Bean
    public GetCurrentGoldPrice getCurrentGoldPrice(ObjectFactory objectFactory) {
        GetCurrentGoldPrice request = objectFactory.createGetCurrentGoldPrice();
        request.setUserName(wsdlProperties.getUser());
        request.setPassword(wsdlProperties.getPassword());
        return request;
    }

}

File: src/main/kotlin/com/thetaris/goldprice/config/WsdlConfig.kt
package com.thetaris.goldprice.config

import com.thetaris.goldprice.generated.GetCurrentGoldPrice
import com.thetaris.goldprice.generated.ObjectFactory
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import javax.validation.constraints.NotNull

@Configuration
class WsdlConfig(private val wsdlProperties: WsdlProperties) {

    @Bean
    fun getCurrentGoldPrice(): GetCurrentGoldPrice {
        val request: GetCurrentGoldPrice = ObjectFactory().createGetCurrentGoldPrice()
        request.userName = wsdlProperties.user
        request.password = wsdlProperties.password
        return request
    }

}

@Configuration
@ConfigurationProperties(prefix = "wsdl")
class WsdlProperties {

    @NotNull
    lateinit var url: String

    @NotNull
    lateinit var action: String

    @NotNull
    lateinit var user: String

    @NotNull
    lateinit var password: String

}
